components/brave_wallet/browser/meld_integration_service.cc

```diff
@@ -0,0 +1,549 @@
+}
+
+void MeldIntegrationService::OnParseServiceProviders(
+    GetServiceProvidersCallback reply_callback,
+    std::optional<std::vector<mojom::MeldServiceProviderPtr>> service_providers)
+    const {
+  if (!service_providers) {
+    std::move(reply_callback)
+        .Run(std::nullopt, std::vector<std::string>{l10n_util::GetStringUTF8(
+                               IDS_WALLET_PARSING_ERROR)});
+    return;
+  }
+  std::move(reply_callback).Run(std::move(service_providers), std::nullopt);
+}
+
+void MeldIntegrationService::GetCryptoQuotes(
+    const std::string& country,
+    const std::string& source_currency_code,
+    const std::string& destination_currency_code,
+    const double source_amount,
+    const std::optional<std::string>& account,
+    GetCryptoQuotesCallback callback) {
+  base::Value::Dict payload;
+  AddKeyIfNotEmpty(&payload, "countryCode", country);
+  AddKeyIfNotEmpty(&payload, "sourceCurrencyCode", source_currency_code);
+  AddKeyIfNotEmpty(&payload, "sourceAmount",
+                   base::NumberToString(source_amount));
+  AddKeyIfNotEmpty(&payload, "destinationCurrencyCode",
+                   destination_currency_code);
+  if (account) {
+    AddKeyIfNotEmpty(&payload, "walletAddress", *account);
+  }
+
+  const std::string json_payload = GetJSON(payload);
+
+  auto url = GURL(base::StringPrintf("%s/payments/crypto/quote",
+                                     GetMeldAssetRatioBaseURL().c_str()));
+  auto internal_callback =
+      base::BindOnce(&MeldIntegrationService::OnGetCryptoQuotes,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  auto conversion_callback = base::BindOnce(&SanitizeJson);
+  api_request_helper_->Request(
+      "POST", url, json_payload, "application/json",
+      std::move(internal_callback), MakeMeldApiHeaders(),
+      {.auto_retry_on_network_change = true, .enable_cache = false},
+      std::move(conversion_callback));
+}
+
+void MeldIntegrationService::OnGetCryptoQuotes(
+    GetCryptoQuotesCallback callback,
+    APIRequestResult api_request_result) const {
+  if (!api_request_result.Is2XXResponseCode() &&
+      !NeedsToParseResponse(api_request_result.response_code())) {
+    std::move(callback).Run(
+        std::nullopt, std::vector<std::string>{
+                          l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR)});
+    return;
+  }
+
+  if (auto errors = ParseMeldErrorResponse(api_request_result.value_body());
+      errors) {
+    std::move(callback).Run(std::nullopt, errors);
+    return;
+  }
+
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      base::BindOnce(&ParseCryptoQuotes, api_request_result.TakeBody()),
+      base::BindOnce(&MeldIntegrationService::OnParseCryptoQuotes,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void MeldIntegrationService::OnParseCryptoQuotes(
+    GetCryptoQuotesCallback reply_callback,
```

Comment:
If there is one callback in current scope it's usually name just as a `callback`